server/src/main/java/org/candlepin/dto/api/v1/ActivationKeyTranslator.java

```diff
@@ -70,58 +71,76 @@ public ActivationKeyDTO populate(ModelTranslator modelTranslator,
             .setServiceLevel(source.getServiceLevel())
             .setAutoAttach(source.isAutoAttach());
 
-        // Process nested objects if we have a model translator to use to the translation...
-        if (modelTranslator != null) {
-            Owner owner = source.getOwner();
-            dest.setOwner(owner != null ? modelTranslator.translate(owner, OwnerDTO.class) : null);
 
-            Set<ActivationKeyPool> pools = source.getPools();
-            if (pools != null && !pools.isEmpty()) {
-                for (ActivationKeyPool poolEntry : pools) {
-                    if (poolEntry != null) {
-                        dest.addPool(new ActivationKeyDTO.ActivationKeyPoolDTO(
-                            poolEntry.getPool().getId(), poolEntry.getQuantity()));
-                    }
+        // Process content overrides
+        Set<ActivationKeyContentOverride> overrides = source.getContentOverrides();
+        dest.setContentOverrides(null);
+
+        if (overrides != null) {
+            Set<ActivationKeyContentOverrideDTO> overrideDTOs = new HashSet<>();
+
+            for (ActivationKeyContentOverride override : overrides) {
+                if (override != null) {
+                    overrideDTOs.add(new ActivationKeyContentOverrideDTO()
+                        .setActivationKey(dest)
+                        .setContentLabel(override.getContentLabel())
+                        .setName(override.getName())
+                        .setValue(override.getValue()));
```

Comment:
This was a tough decision to make, and went through a couple of revisions. At one point in this branch there was an explicit translator for AK content overrides, but it ran into the problem of a circular creation loop since the state between the translators isn't shared. We probably *could* drop this down to a generic content override and use those, but then we're intentionally building incomplete objects, and we'd have to change the collection on the AK DTOs.

After weighing all the pros and cons, I opted for this solution.